nebula-logger/core/main/logger-engine/classes/LogEntryEventBuilder.cls

```diff
@@ -583,9 +584,18 @@ global with sharing class LogEntryEventBuilder {
         String cleanedResponseBody = applyDataMaskRules(response.getBody());
         Boolean responseBodyMasked = cleanedResponseBody != response.getBody();
 
+        String formattedHeadersString;
+        if (LoggerParameter.STORE_HTTP_RESPONSE_HEADER_VALUES == true) {
```

Comment:
MINOR
Please do not compare with literal Booleans. `if(x)` suffices.

Reply:
Yeah, this has come up a few times, including [this discussion](https://github.com/jongpie/NebulaLogger/discussions/383#discussioncomment-3791831) from ~1 year ago. At the moment, Nebula Logger's codebase currently uses these conventions for all `if` statements:

- `if (someCondition == true)`
- `if (someCondition == false)`

Over the years, I've used Nebula Logger's codebase as a place to experiment with different naming conventions, etc, and a few years ago, I settled on using the above convention of explicitly comparing to literal booleans. But today, I don't use this exact convention in any other Apex code that I work on (besides Nebula Logger). Now, I tend to use:

- `if (someCondition)`
- `if (someCondition == false)`

So, I agree that the extra `== true` piece is unnecessary, Nebula Logger. For `false` conditions, I still like to explicitly comparing to the Boolean `false`, I think it's better for human-readability compared to using an `!` for `false`, like `if (!someCondition)`.

With all that said, I think it's finally time to update Nebula Logger's convention to eliminate the extra `== true` bit (and I'll leave the `== false` lines as-is). I can do it fairly easily with a bulk 'find + replace' in VS Code. There are close to 300 matches in 45 Apex classes of `== true` (see screenshot below), so it's technically a lot of lines of code to change, but I think this is a safe change to make / it shouldn't impact any functionality. I'll look at this more today, and will probably include this change in this PR.

Thanks for the feedback!

![image](https://github.com/jongpie/NebulaLogger/assets/1267157/bf40ae7a-6837-4f41-a3ba-27a5a7a3eab7)

Reply:
I can think of one corner case when comparing to a literal Boolean makes sense: by default, all Apex variable are nullable, so actually we can have three values: `false`, `true` and `null`. [The last values causes problems](https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare/). The following snippet will fail with an NPE:
```
Boolean x;
if (x) {}
```
while
```
Boolean x;
if (x == true) {}
```
will not.

I checked how you set `STORE_HTTP_RESPONSE_HEADER_VALUES` and you make sure that it is never `null`. I haven't checked the other occurrences of `x == true` (or false), though.

I'm glad that you are flexible with coding conventions. They are meant to help up, developers, not to be set in stone.

Reply:
Great point about the `null` values corner case - in general, I try to avoid having `null` Boolean values (by always initializing a default value), but I'll test to confirm that `null` won't cause an issue anywhere in the codebase.

Reply:
@TrangOul you were right! There were a few corner cases with some `null` values that caused some test failures - for newer Apex classes, I've been assigning default values, but there were some older classes in the codebase (used for Flow invocables) that didn't have default values assigned.

I updated a few booleans to assign some default values, and everything seems to be working now 🥳 I've pushed those changes to this branch/PR, and I'm hoping to merge this PR today or tomorrow.

Thanks again!